addons/mail/models/mail_render_mixin.py

```diff
@@ -357,53 +371,49 @@ def _render_template_jinja(self, template_txt, model, res_ids,
         if any(r is None for r in res_ids):
             raise ValueError(_('Template rendering should be called on a valid record IDs.'))
 
-        # TDE note: support 'safe' context key as backward compatibility for 6dde919bb9850912f618b561cd2141bffe41340c
-        if options is None:
-            options = {}
-        no_autoescape = options.get('render_safe') or self._context.get('safe')
-
         results = dict.fromkeys(res_ids, u"")
         if not template_txt:
             return results
 
-        # try to load the template
-        try:
-            jinja_env = jinja_safe_template_env if no_autoescape else jinja_template_env
-            template = jinja_env.from_string(tools.ustr(template_txt))
-        except Exception:
-            _logger.info("Failed to load template %r", template_txt, exc_info=True)
-            return results
+        template_instructions = parse_inline_template(str(template_txt))
+        is_dynamic = len(template_instructions) > 1 or template_instructions[0][1]
 
-        if (not self._unrestricted_rendering and template.is_dynamic and not self.env.is_admin() and
+        if (not self._unrestricted_rendering and is_dynamic and not self.env.is_admin() and
            not self.env.user.has_group('mail.group_mail_template_editor')):
             group = self.env.ref('mail.group_mail_template_editor')
             raise AccessError(_('Only users belonging to the "%s" group can modify dynamic templates.', group.name))
 
-        if not template.is_dynamic:
+        if not is_dynamic:
             # Either the content is a raw text without placeholders, either we fail to
             # detect placeholders code. In both case we skip the rendering and return
             # the raw content, so even if we failed to detect dynamic code,
             # non "mail_template_editor" users will not gain rendering tools available
             # only for template specific group users
-            return {record_id: template_txt for record_id in res_ids}
+            return {record_id: template_instructions[0][0] + template_instructions[0][1] for record_id in res_ids}
```

Comment:
Why not just `template_instructions[0][0]` ?

`template_instructions[0][1]` will always be empty for non-dynamic template no ?

Reply:
You're right.